augur/io/metadata.py

```diff
@@ -197,34 +202,64 @@ def read_table_to_dict(table, delimiters, duplicate_reporting=DataErrorMethod.ER
     """
     seen_ids = set()
     duplicate_ids = set()
-    with open_file(table) as handle:
-        # Get sample to determine delimiter
-        table_sample = handle.readline()
+    with open_file(table, "rb") as handle:
+        # open_file(x, "rb") will return x as-is if it's already a file handle,
+        # and in that case the handle might be text mode even though we asked
+        # for bytes.  This assertion guards against usage errors in our caller.
+        assert isinstance(handle.read(0), bytes)
+
+        columns = None
+        records = None
 
+        # Try binary handle as Excel/OpenOffice, as long as it's seekable so we
+        # can reset to the start on failure.
         if handle.seekable():
-            handle.seek(0)
-        else:
-            table_sample_file = StringIO(table_sample)
-            handle = chain(table_sample_file, handle)
+            try:
+                workbook = calamine.load_workbook(handle)
+            except calamine.CalamineError:
+                handle.seek(0)
+            else:
+                rows = workbook.get_sheet_by_index(0).to_python()
+                columns = rows[0]
```

Comment:
_non-blocking_

I tested on a titer Excel file just to see what would happen:

```
$ ./bin/augur curate passthru --metadata ../fludata/VIDRL-Melbourne-WHO-CC/raw-data/A/H3N2/FRA/2023/20230214FRA.xlsx > /dev/null
ERROR: Encountered record with duplicate id '' in '../fludata/VIDRL-Melbourne-WHO-CC/raw-data/A/H3N2/FRA/2023/20230214FRA.xlsx'
```

This is a very specific case where the first non-empty row only had one value, so the columns of the sheet got parsed as 
```
['', 'Focus Reduction Assay ', '', '', '', '', '', '', '', '', '', '', '', '', '', '']
```

We might want to assert that the column values are unique, non-empty strings. (This an issue even outside of Excel files, but just more noticeable here.)

Reply:
Ha, yeah, that file is way beyond this work (though still very parseable with a bespoke program). Since this is an issue for all our inputs, I'm going to leave doing something about it for another time.

Reply:
> a bespoke program

That's the nicest way I've seen fauna described... :D

Reply:
Tracking as future improvement in https://github.com/nextstrain/augur/issues/1553